src/_plugins/author_generator.rb

```diff
@@ -44,8 +44,11 @@ def initialize(site, base, author_dir, author)
       # Set the meta-description for this page.
       meta_description_prefix  = site.config['author_meta_description_prefix'] || 'author: '
       self.data['description'] = "#{meta_description_prefix}#{author}"
-      self.data['publications'] = site.collections.values.flatten.select do |post| post.data["author"] == author end
-      
+      # Posts should be sorted by date descendingly
+      publications = site.collections.values.flatten.select do |post| post.data["author"] == author end
+      publications = publications.sort_by{ |post| post.data["date"] }.reverse
+      self.data['publications'] = publications
```

Comment:
It's probably possible to do this without multiple assignments, something like;

```
self.data['publications'] = site.collections.values.flatten.select 
  do |post| post.data["author"] == author end 
  .sort_by{ |post| post.data["date"] }.reverse
```

It all depends on how Ruby handles newlines, not sure how it does that.

Reply:
Worth a try, I'll try write it as a flow way.